stag-library/src/main/java/com/vimeo/stag/KnownTypeAdapters.java

```diff
@@ -115,7 +115,7 @@ public void write(JsonWriter out, Short value) throws IOException {
         @Override
         public Integer read(JsonReader in) throws IOException {
             try {
-                return in.nextInt();
+                return Integer.valueOf(in.nextInt());
```

Comment:
The warning that Java is giving is to use the primitive type directly instead of casting it into an object  - PFA the details around the warning

![image](https://cloud.githubusercontent.com/assets/13655724/23021724/73d2d278-f473-11e6-9456-ce12be57a3cd.png)


Now looking at the comments and implementation for Integer.valueOf

![image](https://cloud.githubusercontent.com/assets/13655724/23021761/9db4a0d0-f473-11e6-8f49-e30388916d10.png)

Integer, Long, etc are immutable types. So for the purpose of optimisation the platform caches some of these instances so that an allocation need not be done every time. For instance in this case they are caching values between -128 and 127.

Given the number of times this was being called profiling told us that we should use Integer.valueOf instead of the constructor - which is also the recommended way of doing it